src/multirust/command.rs

```diff
@@ -0,0 +1,109 @@
+use std;
+use std::env;
+use std::ffi::OsStr;
+use std::io::{self, Write};
+use std::path::PathBuf;
+use std::process::{self, Command, Output};
+use std::time::Instant;
+
+use Cfg;
+use errors::*;
+use multirust_utils;
+use telemetry;
+use telemetry::TelemetryEvent;
+
+
+pub fn run_command_for_dir<S: AsRef<OsStr>>(cmd: Command,
+                                            args: &[S],
+                                            cfg: &Cfg) -> Result<()> {
+    let arg0 = env::args().next().map(|a| PathBuf::from(a));
+    let arg0 = arg0.as_ref()
+        .and_then(|a| a.file_name())
+        .and_then(|a| a.to_str());
+    let arg0 = try!(arg0.ok_or(Error::NoExeName));
+    if (arg0 == "rustc" || arg0 == "rustc.exe") && cfg.telemetry_enabled() {
+        return telemetry_rustc(cmd, &args, &cfg);
+    }
+    
+    run_command_for_dir_without_telemetry(cmd, &args)
+}
+
+fn telemetry_rustc<S: AsRef<OsStr>>(cmd: Command, args: &[S], cfg: &Cfg) -> Result<()> {
+    let now = Instant::now();
+
+    let output = bare_run_command_for_dir(cmd, &args);
+
+    let duration = now.elapsed();
+
+    let ms = (duration.as_secs() as u64 * 1000) + (duration.subsec_nanos() as u64 / 1000 / 1000);
+
+    match output {
+        Ok(out) => {
+            let exit_code = out.status.code().unwrap_or(1);
+
+            let errors = match out.status.success() {
+                true => None,
+                _ => Some(String::from_utf8_lossy(&out.stderr).to_string()),
+            };
+
+            let _ = io::stdout().write(&out.stdout);
+            let _ = io::stdout().flush();
+            let _ = io::stderr().write(&out.stderr);
+            let _ = io::stderr().flush();
```

Comment:
It looks like the loss of color formatting is a Windows only issue - verified on Ubuntu 15.10 and some version of Fedora (see [Piping stderr strips formatting/colors][1]). 

Windows 10 does support VT100 terminal codes, but after a tentative attempt at enabling them, it looks like `rustup` isn't using the VT100 codes, but is using the older binary codes.

*Possible Fixes*
There are a few possible changes, but they only fix the issue for Windows 10 users. 
* We could use `colored` to deal with terminal colors, but that adds an additional dependency. 
* We could set the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` flag (see retep's comment on [Introducing Colored][2]), but this flag only exists on Windows 10 and it requires ANSI/VT100 color codes.
* Do nothing and inform Windows users that this is a quirk of `rustup` on Windows when they've enabled telemetry.

@brson I think the call on this is up to you. We can always create an issue to fix this after the 1.0 release.

[1]: https://users.rust-lang.org/t/piping-stderr-strips-formatting-colors/5380/6
[2]: https://users.rust-lang.org/t/introducing-colored-the-most-simple-way-to-add-colors-in-your-term/4968/44